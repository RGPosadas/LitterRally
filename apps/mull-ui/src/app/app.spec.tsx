import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import App from './app';

describe('App', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    expect(baseElement).toBeTruthy();
  });

  it('should have a greeting as the title', () => {
    const { getByText } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    expect(getByText('Welcome to mull-ui!')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
