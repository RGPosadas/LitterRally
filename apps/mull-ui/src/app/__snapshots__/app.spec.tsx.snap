// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="main-content"
>
  <div
    className="Toastify"
  />
  <div
    className="nav-container"
  >
    <img
      alt="Mull logo"
      className="logo nav-element"
      src="mull-logo.png"
    />
    <a
      aria-current="page"
      className="active"
      data-testid="home-navlink"
      href="/home"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        className="nav-button mobile-button"
      >
        HomeIcon.svg
      </svg>
    </a>
    <a
      aria-current={null}
      data-testid="map-navlink"
      href="/map"
      onClick={[Function]}
    >
      <svg
        className="nav-button mobile-button"
      >
        MapIcon.svg
      </svg>
    </a>
    <a
      aria-current={null}
      data-testid="create-event-navlink"
      href="/create-event"
      onClick={[Function]}
    >
      <svg
        className="nav-button mobile-middle-button"
      >
        CreateEventIcon.svg
      </svg>
    </a>
    <a
      aria-current={null}
      data-testid="tools-navlink"
      href="/tools"
      onClick={[Function]}
    >
      <svg
        className="nav-button mobile-button"
      >
        ToolMenuIcon.svg
      </svg>
    </a>
    <a
      aria-current={null}
      data-testid="profile-navlink"
      href="/profile"
      onClick={[Function]}
    >
      <svg
        className="nav-button mobile-button"
      >
        ProfileIcon.svg
      </svg>
    </a>
  </div>
</div>
`;
